### PR TITLE
Implement the 'textDocument/rename' command

### PR DESCRIPTION
**Pull request by ccojocar**

### CODE REVIEW

**Comment by ccojocar** [comment]
Thanks! I will try to add some tests in the next days. I see that is possible to spin up a tests langserver.

**Comment by ccojocar** [comment]
@keegancsmith I added some tests for rename command. Please could you have a look? Thanks

**Comment by Dbz** [comment]
Plus 1 for this. I actually just commented in https://github.com/sourcegraph/go-langserver/issues/220

**Comment by ccojocar** [comment]
@keegancsmith Any chance to get this merged soon? Thanks

**Comment by keegancsmith** [comment]
>  ccojocar commented 19 days ago

Shit sorry, did not mean to ignore this. I'll take a look now.

File: langserver/rename.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		},
+	}
+
+	references, err := h.handleTextDocumentReferences(ctx, conn, req, rp)
+	if err != nil {
+		return lsp.WorkspaceEdit{}, err
+	}
+
+	result := lsp.WorkspaceEdit{}
+	if result.Changes == nil {
```
Comment (keegancsmith) [inline langserver/rename.go]: nit: result.Changes will always be nil here. Rather just initialize it

**Pull request merged**